File: paralel/hello.py
# mpirun -np 6 python3 hello.py
from mpi4py import MPI

def main():
    # MPI ortamı mpi4py içe aktarıldığında otomatik başlar (MPI_Init)
    comm = MPI.COMM_WORLD

    # Toplam işlemci sayısını alır (MPI_Comm_size)
    world_size = comm.Get_size()

    # Mevcut işlemcinin numarasını alır (MPI_Comm_rank)
    world_rank = comm.Get_rank()

    # İşlemci adını alır (MPI_Get_processor_name)
    processor_name = MPI.Get_processor_name()

    # Sonucu ekrana yazdır
    print(f"Merhaba! İşlemci: {processor_name}, Rank: {world_rank} / {world_size}")

    # MPI ortamı program bittiğinde otomatik kapanır (MPI_Finalize)

if __name__ == "__main__":
    main()

File: paralel/hello.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char** argv) {
    // MPI ortamını başlat
    MPI_Init(&argc, &argv);

    int dunya_boyutu;
    MPI_Comm_size(MPI_COMM_WORLD, &dunya_boyutu);

    int islemci_sirasi;
    MPI_Comm_rank(MPI_COMM_WORLD, &islemci_sirasi);

    // Her bir işlemci (process) kendi sırasını ve toplam işlemci sayısını yazdırır
    printf("Merhaba! Ben %d. islemciyim. Toplam islemci sayisi: %d\n", islemci_sirasi, dunya_boyutu);

    // MPI ortamını sonlandır
    MPI_Finalize();

    return 0;
}